experimental/packages/opentelemetry-instrumentation-http/test/functionals/utils.test.ts

```diff
@@ -536,4 +537,58 @@ describe('Utility', () => {
       assert.deepStrictEqual(span.attributes['http.request.header.accept'], undefined);
     });
   });
+
+  describe('extractHostnameAndPort', () => {
```

Comment:
These unit tests are great but they only verify that the `extractHostnameAndPort` function is correct.
The original issue discussed that the attribute is missing when request times out or have an error.
I think it could be nice to have a test that asserts the attribute is still present even if request failed. This test should fail before the fix and pass after the PR fix

Reply:
added integration test for it